### PR TITLE
ARROW-10127: Update specification for Decimal to allow for 256-bits

### PR DESCRIPTION
**Pull request by emkornfield**
Will start a thread on the mailing list for a Vote.

### CODE REVIEW

File: format/Schema.fbs [hunk 1]
```diff
@@ -183,10 +183,8 @@ table Decimal {
   /// Number of digits after the decimal point "."
   scale: int;
 
-  /// Number of bits per value. The only accepted width right now is 128 but
-  /// this field exists for forward compatibility so that other bit widths may
-  /// be supported in future format versions. We use bitWidth for consistency
-  /// with Int::bitWidth.
+  /// Number of bits per value. The only accepted widths at 128 and 256.
```
Comment (pitrou) [inline format/Schema.fbs:186]: "are"

**Review by pitrou — APPROVED** [review]
+1 on the principle.

File: format/Schema.fbs [hunk 2]
```diff
@@ -183,10 +183,8 @@ table Decimal {
   /// Number of digits after the decimal point "."
   scale: int;
 
-  /// Number of bits per value. The only accepted width right now is 128 but
```
Comment (pitrou) [inline format/Schema.fbs:186]: Need to update the comments in lines 175-178 above ("Currently only 128-bit (16-byte) integers are used").

File: docs/source/format/CDataInterface.rst [hunk 3]
```diff
@@ -132,21 +132,23 @@ strings:
...
++-----------------+---------------------------------------------------+------------+
+| ``Z``           | large binary                                      |            |
++-----------------+---------------------------------------------------+------------+
+| ``u``           | utf-8 string                                      |            |
++-----------------+---------------------------------------------------+------------+
+| ``U``           | large utf-8 string                                |            |
++-----------------+---------------------------------------------------+------------+
+| ``d:19,10``     | decimal128 [precision 19, scale 10]               |            |
++-----------------+---------------------------------------------------+------------+
+| ``d:19,10,NNN`` | decimal bitwdith = NNN [precision 19, scale 10]   |            |
```
Comment (wesm) [inline docs/source/format/CDataInterface.rst:148]: typo: bit width

Comment (emkornfield) [reply docs/source/format/CDataInterface.rst]: fixed.

File: format/Schema.fbs (hunk 2 above)
Comment (emkornfield) [reply format/Schema.fbs:186]: done.

File: format/Schema.fbs (hunk 1 above)
Comment (emkornfield) [reply format/Schema.fbs]: done.

**Review by jorgecarleitao — APPROVED** [review]
Went through both documents and all looks good to me also. Thanks a lot for driving this, @emkornfield !

**Comment by emkornfield** [comment]
Merging based on vote.

**Pull request closed without merging**